xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/VideoCodec.h

```diff
@@ -0,0 +1,247 @@
+      UPlane,
+      VPlane,
+      MaxPlanes = 3,
+    };
+
+    enum Flags :uint32_t {
+      FLAG_DROP,
+      FLAG_DRAIN
+    };
+
+    VIDEOCODEC_FORMAT videoFormat;
+    uint32_t flags;
+
+    uint32_t width, height;
+
+    uint8_t *decodedData;
+    size_t decodedDataSize;
+
+    uint32_t planeOffsets[VideoPlane::MaxPlanes];
+    uint32_t stride[VideoPlane::MaxPlanes];
+
+    int64_t pts;
+  };
+
+  enum VIDEOCODEC_RETVAL
+  {
+    VC_NONE = 0,        //< noop
+    VC_ERROR,           //< an error occured, no other messages will be returned
+    VC_BUFFER,          //< the decoder needs more data
+    VC_PICTURE,         //< the decoder got a picture
+    VC_EOF,             //< the decoder signals EOF
+  };
+
+  // this are properties given to the addon on create
+  // at this time we have no parameters for the addon
+  typedef struct AddonProps_VideoCodec
+  {
+    int dummy;
+  } AddonProps_VideoCodec;
+
+  struct AddonInstance_VideoCodec;
+  typedef struct KodiToAddonFuncTable_VideoCodec
+  {
+    kodi::addon::CInstanceVideoCodec* addonInstance;
+
+    //! \brief Opens a codec
+    bool (__cdecl* open) (const AddonInstance_VideoCodec* instance, VIDEOCODEC_INITDATA &initData);
+
+    //! \brief Reconfigures a codec
+    bool (__cdecl* reconfigure) (const AddonInstance_VideoCodec* instance, VIDEOCODEC_INITDATA &initData);
+
+    //! \brief Feed codec if requested from GetPicture() (return VC_BUFFER)
+    bool (__cdecl* add_data) (const AddonInstance_VideoCodec* instance, const DemuxPacket &packet);
+
+    //! \brief Get a decoded picture / request new data
+    VIDEOCODEC_RETVAL (__cdecl* get_picture) (const AddonInstance_VideoCodec* instance, VIDEOCODEC_PICTURE &picture);
+
+    //! \brief Get the name of this video decoder
+    const char *(__cdecl* get_name) (const AddonInstance_VideoCodec* instance);
+
+    //! \brief Reset the codec
+    void (__cdecl* reset)(const AddonInstance_VideoCodec* instance);
+  } KodiToAddonFuncTable_VideoCodec;
+
+  typedef struct AddonToKodiFuncTable_VideoCodec
+  {
+    KODI_HANDLE kodiInstance;
+    bool(*get_frame_buffer)(void* kodiInstance, VIDEOCODEC_PICTURE &picture);
+  } AddonToKodiFuncTable_VideoCodec;
+
+  typedef struct AddonInstance_VideoCodec
+  {
+    AddonProps_VideoCodec props;
+    AddonToKodiFuncTable_VideoCodec toKodi;
+    KodiToAddonFuncTable_VideoCodec toAddon;
+  } AddonInstance_VideoCodec;
+}
+
+namespace kodi
+{
+  namespace addon
+  {
+
+    class CInstanceVideoCodec : public IAddonInstance
+    {
+    public:
+      CInstanceVideoCodec(KODI_HANDLE instance)
+        : IAddonInstance(ADDON_INSTANCE_VIDEOCODEC)
+      {
+        if (CAddonBase::m_interface->globalSingleInstance != nullptr)
+          throw std::logic_error("kodi::addon::CInstanceVideoCodec: Creation of multiple together with single instance way is not allowed!");
+
+        SetAddonStruct(instance);
+      }
+
+      virtual ~CInstanceVideoCodec() { }
+
+      //! \copydoc CInstanceVideoCodec::Open
```

Comment:
Only a note for the time where I reworking the docs and no need to change something.

Want to have the docs complete here on headers without use of `\copydoc`, also place this to the same place :) thing you mean `\copydoc KodiToAddonFuncTable_VideoCodec::open`